pkg/client/typed/dynamic/client_pool.go

```diff
@@ -28,16 +29,21 @@ import (
 
 // ClientPool manages a pool of dynamic clients.
 type ClientPool interface {
```

Comment:
Can you add a comment to clientPoolImpl saying it reuses restclient for resources/kind of the same group version? It's less obvious after the interface changes.